### ISSUE TITLE
Seed phrase addresses

### ISSUE THREAD

**Issue by trymeouteh** [simplex-chat/simplex-chat#2167]
I would like to suggest doing what bitcoin and monero does is use a seed phrase. The 24 word seed phrase can be easily backed up and restored like any other crypto wallet and the seed phrase can generate millions of long term addresses.

https://iancoleman.io/bip39/

Instead of having one-time addresses, each account will have a seed phrase which the user can backup by writing down or storing in a password manager. This way if a user has not made a backup and loses their device, they can still recocer their account on another device without a backup file by using their seed phrase which will restore all.of their addresses which will restore their conversations/connections. Of course te seed phrase will not restore the labels you saved on your contacts and other settings which is why the file backup feature should still be kept.

It will also be wise to add a way to label on each address in SimpleX which can be imported and exported with a SimpleX backup file to help the user know what each address is for.

And it will also be wise to enable/disable addresses. If a is no longer used and is constantly getting spammed like an email address that was sold on the dark web, the user can just disable the address and no longer be bothered by spam.

I do not think addresses should be deleted but instead disabled.

The seed phrase address generation used by crypto ensures addresses by the usere can be recovered with a one time single backup of 24 words.

**Comment by epoberezkin** [comment]
The peace of having this feature is the lack of (or limited) privacy and centralisation of public blockchains. There is no centralised state or global addressing scheme in SimpleX network that would be required to restore bigger state from a smaller passphrase.

**Comment by trymeouteh** [comment]
No blockchain is required since the addresses are generated by a seed phrase and can be recovered with a 24 word seed phrase. The seed phrase recovery and generation can be done offline.

**Comment by epoberezkin** [comment]
For information to be recoverable from somewhere, it has to be stored somewhere to begin with. You cannot just inflate a smaller seed phrase into a much large state, without that state being retrieved from somewhere – seed is used to authorise the access, and to decrypt the data, but it doesn't contain the actual data.

In case of BTC/etc. this somewhere is public blockchain. Where in your view the information about SimpleX Chat connections would be stored for them to be recoverable from seed phrase?

**Comment by epoberezkin** [comment]
You might mean a flipped approach when both queue addresses and access keys are generated client-side from a single seed. As opposed to the current approach when addresses are generated by relays and keys are generated randomly, using system entropy sources where available (iOS doesn't give access to them, for example) or some cryptographically strong sources of randomness.

This flipped approach is possible, but:
- this is not how the whole system is designed, so it's not an additive change - it's a full ground up re-design.
- this state won't include any meta-data received from the contacts (so once the connection is restored the profile will have to be re-sent).
- client-generated addresses may have collisions on the relays (which can be worked around as well).
- this approach is less secure than randomly generated addresses and keys - once an attacker gets the seed they may continue generating addresses and access keys.
- it won't work with double ratchet at all, as it fundamentally has external source of randomness - another contact, so either it would require re-negotiating ratchet anyway, or it would break both break-in recovery and forward secrecy.

So while seed-based approach is common and convenient, it's less secure. A better approach is to keep the state on multiple devices (to avoid losing it), without relying on memorised seed.

**Comment by trymeouteh** [comment]
Those are interesting points.

I lile the idea of a seed phrase since it allowed the user to backup 24 words which is a one time backup. And if they lose their devices and did not do a backup for lets say over a year, they can still restore their addresses.

The backup file is more secure but requires to be backed up every time an connection is made to ensure you do not lose a single contact.

Will there be a robust backup feature in the future which will do automatic backups of your contacts weather your on mobile or desktop everytime a new connection is made? And could these automatic backups be created and stored on the file system or the cloud such as Filen, Nextcloud, etc?

I am pretty good with keeping backups, but lots of people are not and a messanging app, you will likely be communication with others who do not backup their data as often and if they are in another country and the break their device, communication is lost.

**Comment by trymeouteh** [comment]
How do contact address work? Essentially seed phrase addresses is the exact same as contact addresses except the keys are generated locally with a seed phrase.

**Comment by Mikaela** [comment]
This is how I would like to see #444 as an user

**Comment by azazar** [comment]
> For information to be recoverable, it has to be stored somewhere to begin with. You cannot just inflate a smaller seed phrase into a much large state, without that state being retrieved from somewhere – seed is used to authorise the access, and to decrypt the data, but it doesn't contain the actual data.
> 
> In case of BTC/etc. this somewhere is public blockchain. Where in your view the information about SimpleX Chat connections would be stored for them to be recoverable from seed phrase?

Wrong. You don't need to store or recover information when you can deterministically derive it. A small seed can be inflated into a much larger state using pseudo-random functions—no external storage required. This is the foundation of HD wallets and other cryptographic systems: the seed generates keys and identities on demand, not by retrieving them, but by computing them.

**Comment by hippi777** [comment]
any computer user should be expected to be able to create a digital backup of whatever matters to them, and this seed stuff is kinda pointless... however my question is whether a backup could become corrupted if done at the wrong moment while simplex is running? any timed backup solution could be affected, and i would hack around such like wrapping the executable to make a copy anywhere before/after running simplex, so a restart after anything important could give me an inactive state that couldnt become corrupted... overwriting an old but good backup with a new but broken backup is serious...

otherwise exporting (manual or auto on whatever strategy, where tags could be useful) and signaling important events like making a new contact (while the conversations are really less important) could take the essence of the topic forward...

having solid support for backups should be enough, anything beyond sounds like overengineering of simplex...

(ps: still sorry if i question anything obvious from the pov of an actual user cuz my unsupported platforms)
